native/acter/src/api/device.rs

```diff
@@ -90,21 +95,19 @@ impl DeviceController {
                     break;
                 }
                 if let Some(user_devices) = device_updates.new.get(my_id) {
-                    for device in user_devices.values() {
-                        let dev_id = device.device_id().to_owned();
+                    for (dev_id, dev) in user_devices {
                         info!("device-new device id: {}", dev_id);
-                        let evt = DeviceNewEvent::new(&client, dev_id);
-                        if let Err(e) = new_event_tx.try_send(evt) {
+                        let evt = DeviceNewEvent::new(&client, dev_id.to_owned());
+                        if let Err(e) = new_tx.try_send(evt) {
                             error!("Dropping devices new event: {}", e);
                         }
                     }
                 }
                 if let Some(user_devices) = device_updates.changed.get(my_id) {
-                    for device in user_devices.values() {
-                        let dev_id = device.device_id().to_owned();
+                    for (dev_id, dev) in user_devices {
                         info!("device-changed device id: {}", dev_id);
-                        let evt = DeviceChangedEvent::new(&client, dev_id);
-                        if let Err(e) = changed_event_tx.try_send(evt) {
+                        let evt = DeviceChangedEvent::new(&client, dev_id.to_owned());
+                        if let Err(e) = changed_tx.try_send(evt) {
```

Comment:
reading the subscriber on the flutter side, there are two clear improvements here:
1. we don't care about the details of the event on the flutter side, all that happens is that we refresh the entire list
2. it's not needed to be done _for each_ device that changed, but once for every change reported. That's totally enough.

This can and should be vastly simplified.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1598/commits/943d0005f535bc3b169e251f3c41dc83aabd8732